wire-protoc-integration/src/main/java/com/squareup/wire/protocwire/JavaTargetConfig.kt

```diff
@@ -0,0 +1,36 @@
+package com.squareup.wire.protocwire
+
+import com.squareup.wire.schema.JavaTarget
+import com.squareup.wire.gradle.JavaOutput
+
+private const val ANDROID = "android"
+private const val ANDROID_ANNOTATIONS = "androidAnnotations"
+private const val COMPACT = "compact"
+private const val EMIT_DECLARED_OPTIONS = "emitDeclaredOptions"
+private const val EMIT_APPLIED_OPTIONS = "emitAppliedOptions"
+private const val BUILDERS_ONLY = "buildersOnly"
+
+/**
+ * Translates the protoc options into known options for the
+ * JavaTarget via the gradle plugin JavaOutput.
+ */
+internal class JavaTargetConfig {
+  companion object {
+    /**
+     * @param parameters the map of string to string from the protoc options.
+     * @return the translated JavaTarget.
+     */
+    internal fun parse(parameters: Map<String, String>): JavaTarget {
```

Comment:
protoc doesn't have a type wrapping this map? That'd be sweet if we could do something like
`JavaTarget.of(optionMap)`. I'm not sure how we can add this API without touching `JavaTarget` directly.

If we add an empty companion on `JavaTarget`, we might be able to do so. I don't know how to set it `JvmStatic` so the java call site would not be nice.
![image](https://user-images.githubusercontent.com/1767669/218969009-d71ad358-7536-4e29-afc9-cd02142019d2.png)

Reply:
Yeah, I was considering that but I was erroring on keeping the changes in an isolated module here. I can make the change to move it to the `JavaTarget` and `KotlinTarget` if that is preferred